igvm/hypervisor.py

```diff
@@ -57,7 +57,7 @@ class Hypervisor(Host):
     def __init__(self, dataset_obj):
         super(Hypervisor, self).__init__(dataset_obj)
 
-        if dataset_obj['state'] == 'retired':
+        if dataset_obj['state'] not in ['online', 'online_reserved']:
```

Comment:
The maintenance and deploy states must still allow to create the object. This is needed to be able to connect to a hypervisor to move VMs out of it.

Reply:
In `deploy_online` nothing should be happening on the hypervisor, but for maintenance it makes sense.